console/console-init/ui/src/components/PasswordInputFieldWithToggle/PasswordInputFieldWithToggle.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import { TextInput, TextInputProps } from "@patternfly/react-core";
+import { EyeIcon, EyeSlashIcon } from "@patternfly/react-icons";
+import { StyleSheet } from "@patternfly/react-styles";
+
+const styles = StyleSheet.create({
+  icon: {
+    minWidth: 35
+  },
+  textInput: {
+    marginRight: -35
+  }
+});
+
+export interface IPasswordInputFieldWithToggleProps extends TextInputProps {
+  id: string;
+}
+
+export const PasswordInputFieldWithToggle: React.FC<IPasswordInputFieldWithToggleProps> = ({
+  id,
+  onChange,
+  name,
+  isValid = true,
+  validated
+}) => {
+  const [isShowPassword, setIsShowPassword] = useState<boolean>(false);
+
+  const onToggle = (isShowPassword: boolean) => {
+    setIsShowPassword(isShowPassword);
+  };
+
+  const renderIcon = () => {
+    if (isValid) {
+      if (isShowPassword) {
+        return (
+          <EyeSlashIcon
+            className={styles.icon}
+            onClick={() => onToggle(false)}
+          />
+        );
+      }
+      return <EyeIcon className={styles.icon} onClick={() => onToggle(true)} />;
+    }
+  };
+
+  const type = isShowPassword === true ? "text" : "password";
```

Comment:
Can we omit the comparision of isShowPassword with true, seems bit redundant.

Reply:
Sorry, I don't get your point what you want. can you please describe little bit in detail.

Reply:
I meant `if(x)` and `if(x === true)` evaluate to be the same.